server/controller/src/main/java/ai/starwhale/mlops/common/DockerImage.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2022 Starwhale, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.starwhale.mlops.common;
+
+import ai.starwhale.mlops.exception.SwValidationException;
+import ai.starwhale.mlops.exception.SwValidationException.ValidSubject;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.springframework.util.StringUtils;
+
+@Getter
+@ToString
+@EqualsAndHashCode
+@AllArgsConstructor
+@NoArgsConstructor
+public class DockerImage {
+
+    String registry;
+    String image;
+
+    private static final Pattern PATTERN_IMAGE_FULL = Pattern.compile(
+            "^(([a-z0-9-]+\\.[a-z0-9.]+)(:(\\d{2,5}))?)?\\/?(.+)$");
```

Comment:
https://stackoverflow.com/questions/37861791/how-are-docker-image-names-parsed
> TL;DR: The hostname must contain a . dns separator, a : port separator, or the value "localhost" before the first /. Otherwise the code assumes you want the default registry, Docker Hub.

1. Do we support the `localhost` domain?
2. Do we validate the original docker image name? 
    * If not, should we just split the image name with a slash and check if the first part is a valid domain?

Reply:
done